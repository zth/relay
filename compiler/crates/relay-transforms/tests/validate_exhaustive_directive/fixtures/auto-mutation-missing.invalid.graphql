# expected-to-throw
mutation AutoMissing {
  unionField {
    ... on A { __typename }
  }
}
# %extensions%
extend type Mutation { unionField: MyUnion }
type A { id: ID }
type B { id: ID }
union MyUnion = A | B
