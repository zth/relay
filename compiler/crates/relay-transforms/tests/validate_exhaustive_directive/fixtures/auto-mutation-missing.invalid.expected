==================================== INPUT ====================================
# expected-to-throw
mutation AutoMissing {
  unionField {
    ... on A { __typename }
  }
}
# %extensions%
extend type Mutation { unionField: MyUnion }
type A { id: ID }
type B { id: ID }
union MyUnion = A | B
==================================== ERROR ====================================
✖︎ Field 'unionField' marked with @exhaustive is missing selection for union members: 'B'.

  auto-mutation-missing.invalid.graphql:3:3
    2 │ mutation AutoMissing {
    3 │   unionField {
      │   ^^^^^^^^^^
    4 │     ... on A { __typename }
